src/Microsoft.Tye.Core/DockerContainerBuilder.cs

```diff
@@ -34,55 +34,93 @@ public static async Task BuildContainerImageAsync(OutputContext output, Applicat
-            // different depending on that choice.
-            string contextDirectory;
-            if (container.UseMultiphaseDockerfile ?? true)
-            {
-                contextDirectory = ".";
-            }
-            else
+            TempFile? tempFile = null;
+            TempDirectory? tempDirectory = null;
+
+            try
             {
-                var publishOutput = project.Outputs.OfType<ProjectPublishOutput>().FirstOrDefault();
-                if (publishOutput is null)
+                // We need to know if this is a single-phase or multi-phase Dockerfile because the context directory will be
+                // different depending on that choice.
+                //
+                // For the cases where generate a Dockerfile, we have the constraint that we need
+                // to place it on the same drive (Windows) as the docker context.
+                if (container.UseMultiphaseDockerfile ?? true)
+                {
+                    // For a multi-phase Docker build, the context is always the project directory.
+                    contextDirectory = ".";
+
+                    if (File.Exists(dockerFilePath))
+                    {
+                        output.WriteDebugLine($"Using existing Dockerfile '{dockerFilePath}'.");
+                    }
+                    else
+                    {
+                        // We need to write the file, let's stick it under obj.
+                        Directory.CreateDirectory(Path.Combine(project.ProjectFile.DirectoryName, "obj"));
+                        dockerFilePath = Path.Combine(project.ProjectFile.DirectoryName, "obj", "Dockerfile");
+
+                        // Clean up file when done building image
+                        tempFile = new TempFile(dockerFilePath);
+
+                        await DockerfileGenerator.WriteDockerfileAsync(output, application, project, container, tempFile.FilePath);
+                    }
+                }
+                else
+                {
+                    // For a single-phase Docker build the context is always the directory containing the publish
+                    // output. We need to put the Dockerfile in the context directory so it's on the same drive (Windows).
+                    var publishOutput = project.Outputs.OfType<ProjectPublishOutput>().FirstOrDefault();
+                    if (publishOutput is null)
+                    {
+                        throw new InvalidOperationException("We should have published the project for a single-phase Dockerfile.");
+                    }
+
+                    contextDirectory = publishOutput.Directory.FullName;
+
+                    // Clean up directory when done building image
+                    tempDirectory = new TempDirectory(publishOutput.Directory);
+
+                    if (File.Exists(dockerFilePath) & container.UseMultiphaseDockerfile == false)
```

Comment:
&& or &?

Reply:
ooh wow, that's a rare one. Should be `&&`